Integrate with Angular

I tried to integrate this component in angular (16). I started creating a empty Project with started sample for the regl-scatterplot component. When try start the project I obtained these erros:

```
Warning: /home/miguel/Sources/my-regl-scatterplot/node_modules/regl-scatterplot/dist/regl-scatterplot.esm.js depends on 'regl'. CommonJS or AMD dependencies can cause optimization bailouts.
For more info see: https://angular.io/guide/build#configuring-commonjs-dependencies

Error: node_modules/regl-scatterplot/dist/types.d.ts:21:21 - error TS2307: Cannot find module 'd3-scale' or its corresponding type declarations.

21 type Scale = import('d3-scale').ScaleContinuousNumeric<number, number>;
                       ~~~~~~~~~~
Error: node_modules/regl-scatterplot/dist/types.d.ts:140:38 - error TS2307: Cannot find module './renderer' or its corresponding type declarations.

140   renderer: ReturnType<typeof import('./renderer').createRenderer>;
```

The first one was resolved installing the D3 Types:

`npm i @types/d3-scale --save-dev`

But the second one persist ` Cannot find module './renderer' `. This is where exist this problem:

```
export type Properties = {
  **renderer: ReturnType<typeof import('./renderer').createRenderer>;**
  canvas: HTMLCanvasElement;
  regl: import('regl').Regl;
  syncEvents: boolean;
  version: string;
  lassoInitiatorElement: HTMLElement;
  camera: Camera2D;
  performanceMode: boolean;
  opacityByDensityDebounceTime: number;
  points: [number, number][];
  pointsInView: number[];
  isDestroyed: boolean;
  isPointsDrawn: boolean;
  isPointsFiltered: boolean;
  hoveredPoint: number;
  filteredPoints: number[];
  selectedPoints: number[];
} & Settable;
```

I don't have any idea to resolve it. Do you have some idea how fix this problem?

--- a/.github/workflows/build.yml
+++ b/.github/workflows/build.yml
@@ -8,7 +8,7 @@ on:
 
 jobs:
   build:
-    runs-on: macos-latest
+    runs-on: ubuntu-latest
 
     strategy:
       matrix:
@@ -26,4 +26,5 @@ jobs:
       - name: Build regl-scatterplot
         run: npm run build --if-present
       - name: Test regl-scatterplot
-        run: npm test
+        run: xvfb-run npm test
+        timeout-minutes: 5

--- a/.github/workflows/build.yml
+++ b/.github/workflows/build.yml
@@ -9,22 +9,23 @@ on:
 jobs:
   build:
     runs-on: ubuntu-latest
-
-    strategy:
-      matrix:
-        node-version: [18.x]
-
     steps:
-      - name: Checkout
-        uses: actions/checkout@v4
-      - name: Use Node.js ${{ matrix.node-version }}
-        uses: actions/setup-node@v4
+      - uses: actions/checkout@v4
+      - uses: actions/setup-node@v4
         with:
-          node-version: ${{ matrix.node-version }}
-      - name: Install regl-scatterplot
-        run: npm ci
-      - name: Build regl-scatterplot
-        run: npm run build --if-present
-      - name: Test regl-scatterplot
-        run: xvfb-run npm test
+          node-version: 18
+      - run: npm ci
+      - run: npm run build --if-present
+        env:
+          CI: true
+
+  test:
+    runs-on: macos-latest
+    steps:
+      - uses: actions/checkout@v4
+      - uses: actions/setup-node@v4
+        with:
+          node-version: 18
+      - run: npm ci
+      - run: npm test
         timeout-minutes: 5

--- a/.github/workflows/publish.yml
+++ b/.github/workflows/publish.yml
@@ -7,7 +7,7 @@ on:
 
 jobs:
   build:
-    runs-on: macos-latest
+    runs-on: ubuntu-latest
     steps:
       - uses: actions/checkout@v4
       - uses: actions/setup-node@v4
@@ -20,7 +20,7 @@ jobs:
 
   publish-npm:
     needs: build
-    runs-on: macos-latest
+    runs-on: ubuntu-latest
     steps:
       - uses: actions/checkout@v4
       - uses: actions/setup-node@v4

--- a/.github/workflows/release.yml
+++ b/.github/workflows/release.yml
@@ -8,7 +8,7 @@ on:
 jobs:
   build:
     name: Create Release
-    runs-on: macos-latest
+    runs-on: ubuntu-latest
     steps:
       - name: Checkout code
         uses: actions/checkout@v4

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -1,3 +1,8 @@
+## 1.8.5
+
+- Fix: add `d3-scale` types as a dev dependency [#153](https://github.com/flekschas/regl-scatterplot/issues/153)
+- Fix: run CI workflows on Ubuntu to ensure the `sed` replacement call works properly [#153](https://github.com/flekschas/regl-scatterplot/issues/153)
+
 ## 1.8.4
 
 - Fix a regression with the point opacity introduced in [v1.7.1](#1.7.1)

--- a/package.json
+++ b/package.json
@@ -57,6 +57,7 @@
     "@rollup/plugin-json": "^6.0.0",
     "@rollup/plugin-node-resolve": "^15.1.0",
     "@rollup/plugin-terser": "^0.4.3",
+    "@types/d3-scale": "^4.0.6",
     "@types/node": "^20.3.1",
     "acorn": "^8.9.0",
     "apache-arrow": "^12.0.1",

